## Import Libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

## Read Data

In [2]:
df = pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG.csv')

In [3]:
df

Area Code      Area  Item Code           Item  Element Code  \
0              4   Algeria       2501     Population           511   
1              4   Algeria       2501     Population          5301   
2              4   Algeria       2901    Grand Total           664   
3              4   Algeria       2901    Grand Total           674   
4              4   Algeria       2901    Grand Total           684   
...          ...       ...        ...            ...           ...   
60938        181  Zimbabwe       2899  Miscellaneous          5142   
60939        181  Zimbabwe       2899  Miscellaneous           645   
60940        181  Zimbabwe       2899  Miscellaneous           664   
60941        181  Zimbabwe       2899  Miscellaneous           674   
60942        181  Zimbabwe       2899  Miscellaneous           684   

                                      Element             Unit     Y2014  \
0               Total Population - Both sexes     1000 persons  38924.00   
1                    Domestic supply quantity      1000 tonnes      0.00   
2               Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3      Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4          Fat supply quantity (g/capita/day)     g/capita/day     80.06   
...                                       ...              ...       ...   
60938                                    Food      1000 tonnes     42.00   
60939     Food supply quantity (kg/capita/yr)               kg      3.06   
60940           Food supply (kcal/capita/day)  kcal/capita/day      3.00   
60941  Protein supply quantity (g/capita/day)     g/capita/day      0.10   
60942      Fat supply quantity (g/capita/day)     g/capita/day      0.04   

          Y2015     Y2016     Y2017     Y2018  
0      39728.00  40551.00  41389.00  42228.00  
1          0.00      0.00      0.00      0.00  
2       3379.00   3372.00   3341.00   3322.00  
3         94.35     94.72     92.82     91.83  
4         79.36     77.40     80.19     77.28  
...         ...       ...       ...       ...  
60938     46.00     33.00     19.00     16.00  
60939      3.33      2.35      1.33      1.08  
60940      4.00      3.00      1.00      1.00  
60941      0.11      0.08      0.04      0.04  
60942      0.05      0.03      0.02      0.01  

[60943 rows x 12 columns]

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 60943 entries, 0 to 60942
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     60943 non-null  int64  
 1   Area          60943 non-null  object 
 2   Item Code     60943 non-null  int64  
 3   Item          60943 non-null  object 
 4   Element Code  60943 non-null  int64  
 5   Element       60943 non-null  object 
 6   Unit          60943 non-null  object 
 7   Y2014         59354 non-null  float64
 8   Y2015         59395 non-null  float64
 9   Y2016         59408 non-null  float64
 10  Y2017         59437 non-null  float64
 11  Y2018         59507 non-null  float64
dtypes: float64(5), int64(3), object(4)
memory usage: 5.6+ MB


In [4]:
df.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [5]:
df.describe(include='all')

Area Code   Area     Item Code                     Item  \
count   60943.000000  60943  60943.000000                    60943   
unique           NaN     49           NaN                      119   
top              NaN  Kenya           NaN  Milk - Excluding Butter   
freq             NaN   1560           NaN                     1262   
mean      134.265576    NaN   2687.176706                      NaN   
std        72.605709    NaN    146.055739                      NaN   
min         4.000000    NaN   2501.000000                      NaN   
25%        74.000000    NaN   2562.000000                      NaN   
50%       136.000000    NaN   2630.000000                      NaN   
75%       195.000000    NaN   2775.000000                      NaN   
max       276.000000    NaN   2961.000000                      NaN   

        Element Code                   Element         Unit          Y2014  \
count   60943.000000                     60943        60943   59354.000000   
unique           NaN                        18            5            NaN   
top              NaN  Domestic supply quantity  1000 tonnes            NaN   
freq             NaN                      5295        40933            NaN   
mean     3814.856456                       NaN          NaN     134.196282   
std      2212.007033                       NaN          NaN    1567.663696   
min       511.000000                       NaN          NaN   -1796.000000   
25%       684.000000                       NaN          NaN       0.000000   
50%      5142.000000                       NaN          NaN       0.090000   
75%      5511.000000                       NaN          NaN       8.340000   
max      5911.000000                       NaN          NaN  176405.000000   

                Y2015          Y2016          Y2017          Y2018  
count    59395.000000   59408.000000   59437.000000   59507.000000  
unique            NaN            NaN            NaN            NaN  
top               NaN            NaN            NaN            NaN  
freq              NaN            NaN            NaN            NaN  
mean       135.235966     136.555222     140.917765     143.758381  
std       1603.403984    1640.007194    1671.862359    1710.782658  
min      -3161.000000   -3225.000000   -1582.000000   -3396.000000  
25%          0.000000       0.000000       0.000000       0.000000  
50%          0.080000       0.080000       0.100000       0.070000  
75%          8.460000       8.430000       9.000000       9.000000  
max     181137.000000  185960.000000  190873.000000  195875.000000

In [6]:
df.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

In [11]:
pd.set_option('display.max_rows',None)

## What is the total sum of Animal Fat Produced in 2014 and 2017 respectively?

In [15]:
#perform a groupby sum aggregration on 'Item'

df.groupby('Item').sum()

Area Code  Item Code  Element Code  \
Item                                                                  
Alcohol, Non-Food                    38447     752497       1515090   
Alcoholic Beverages                  75016    1649136       2171381   
Animal Products                      18060     397035         90990   
Animal fats                          85279    1882494       2567303   
Apples and products                  69280    1352989       1920418   
Aquatic Animals, Others              47420     966381       1139179   
Aquatic Plants                       51178    1057275       1278386   
Aquatic Products, Other              52570    1160712       1320249   
Bananas                              68496    1338880       1894890   
Barley and products                  76321    1432410       2243002   
Beans                                78001    1476680       2263509   
Beer                                 65975    1314720       1813728   
Beverages, Alcoholic                 65918    1313052       1807946   
Beverages, Fermented                 65203    1299273       1774758   
Bovine Meat                          70033    1431044       1963251   
Butter, Ghee                         60516    1238480       1594919   
Cassava and products                 73208    1385004       2081916   
Cephalopods                          58593    1233636       1571763   
Cereals - Excluding Beer             95370    2074170       2976719   
Cereals, Other                       79505    1489320       2326237   
Citrus, Other                        70239    1367122       1954812   
Cloves                               53398    1064726       1420590   
Cocoa Beans and products             70862    1398123       1998287   
Coconut Oil                          69418    1330248       1959863   
Coconuts - Incl Copra                74263    1410560       2104761   
Coffee and products                  72724    1433350       2070618   
Cottonseed                           44285     841911       1768564   
Cottonseed Oil                       67319    1287500       1881106   
Cream                                55778    1141088       1421946   
Crustaceans                          64831    1338260       1761859   
Dates                                64881    1278072       1771007   
Demersal Fish                        62436    1292616       1677632   
Eggs                                154150    3250703       4426246   
Fats, Animals, Raw                   79842    1636726       2341974   
Fish, Body Oil                       56619    1154115       1462871   
Fish, Liver Oil                      48755     979264       1197204   
Fish, Seafood                        71756    1583600       2039247   
Freshwater Fish                      64920    1333563       1761607   
Fruits - Excluding Wine              79433    1725129       2313131   
Fruits, Other                        76833    1506750       2224160   
Grand Total                          18060     391635         90990   
Grapefruit and products              69139    1340469       1896997   
Grapes and products (excl wine)      70255    1367640       1946827   
Groundnut Oil                        70147    1355444       1971148   
Groundnuts (Shelled Eq)              82299    1577052       2457038   
Honey                                69241    1413675       1908893   
Infant food                          59745    1197960       1547996   
Lemons, Limes and products           70391    1358240       1938946   
Maize Germ Oil                       64256    1234196       1747258   
Maize and products                   91790    1727118       2832303   
Marine Fish, Other                   64234    1323956       1733842   
Meat                                 72011    1583334       2037234   
Meat, Other                          69390    1416730       1930240   
Milk - Excluding Butter             168482    3657276       5058264   
Millet and products                  77725    1401969    

### Result: 
    Total sum of Animal fats produced in 2014 is 209460.54
    Total sum of Animal fats produced in 2017 is 269617.53

In [14]:
df.groupby('Item').get_group('Animal fats').sum()

Area Code                                                   85279
Area            AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...
Item Code                                                 1882494
Item            Animal fatsAnimal fatsAnimal fatsAnimal fatsAn...
Element Code                                              2567303
Element         ProductionImport QuantityStock VariationExport...
Unit            1000 tonnes1000 tonnes1000 tonnes1000 tonnes10...
Y2014                                                   209460.54
Y2015                                                   200675.72
Y2016                                                   183314.22
Y2017                                                   269617.53
Y2018                                                   269648.27
dtype: object

## What is the mean and Standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [16]:
df.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

### Result:
    The mean across the whole dataset for year 2015 is 135.236
    The standard deviation across the whole dataset for year 2015 is 1603.404


## What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [19]:
df.isnull().sum()

Area Code          0
Area               0
Item Code          0
Item               0
Element Code       0
Element            0
Unit               0
Y2014           1589
Y2015           1548
Y2016           1535
Y2017           1506
Y2018           1436
dtype: int64

In [25]:
for col in df.columns:
    percentage_missing = np.mean(df[col].isnull())*100
    print('{} - {}%'.format(col,percentage_missing))

Area Code - 0.0%
Area - 0.0%
Item Code - 0.0%
Item - 0.0%
Element Code - 0.0%
Element - 0.0%
Unit - 0.0%
Y2014 - 2.6073544131401474%
Y2015 - 2.5400784339464746%
Y2016 - 2.5187470259094566%
Y2017 - 2.4711615772114928%
Y2018 - 2.356300149319856%


### Result:
    The total number and percentage of missing data in 2016 is 1535 and 2.52%

## Which year had the highest correlation with 'Element Code'?

In [35]:
df.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

In [36]:
df.corr()*100

Area Code   Item Code  Element Code       Y2014       Y2015  \
Area Code     100.000000   -0.515910     -0.020860    0.616411    0.547166   
Item Code      -0.515910  100.000000     -2.468327    2.172208    2.085737   
Element Code   -0.020860   -2.468327    100.000000    2.445682    2.388920   
Y2014           0.616411    2.172208      2.445682  100.000000   99.464736   
Y2015           0.547166    2.085737      2.388920   99.464736  100.000000   
Y2016           0.524714    2.010913      2.344400   99.608122   99.573892   
Y2017           0.500578    2.149439      2.425401   99.522984   98.804787   
Y2018           0.566540    2.131369      2.427857   99.487166   98.820845   

                   Y2016       Y2017       Y2018  
Area Code       0.524714    0.500578    0.566540  
Item Code       2.010913    2.149439    2.131369  
Element Code    2.344400    2.425401    2.427857  
Y2014          99.608122   99.522984   99.487166  
Y2015          99.573892   98.804787   98.820845  
Y2016         100.000000   99.278463   99.275667  
Y2017          99.278463  100.000000   99.810327  
Y2018          99.275667   99.810327  100.000000

### Result:
    2014 (Y2014) had the highest correlation with Element Code of 2.45%

## What year had the highest sum of Import Quantity?

In [40]:
df.groupby('Element').sum()

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                   708993   14197445      28068795   
Export Quantity                            599910   11840553      26026133   
Fat supply quantity (g/capita/day)         675050   13535000       3435732   
Feed                                       176272    3538507       7282199   
Food                                       663295   13285035      25406622   
Food supply (kcal/capita/day)              674057   13511060       3329296   
Food supply quantity (kg/capita/yr)        658446   13185401       3163725   
Import Quantity                            688174   13795966      28834929   
Losses                                     274353    5424803      10292107   
Other uses (non-food)                      235554    4729749       8926728   
Processing                                 271940    5350416      10313310   
Production                                 526751   10450053      21388191   
Protein supply quantity (g/capita/day)     675050   13535000       3385502   
Residuals                                  623271   12421089      24066350   
Seed                                       103537    2035933       4211574   
Stock Variation                            571566   11329527      21464704   
Total Population - Both sexes                6020     112545         22995   
Tourist consumption                         50308    1486528       2869905   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein 

### Result:
    2017 (Y2017, 294559.09) has the highest sum of Import Quantity

## What is the total number of the sum of production in 2014?

In [41]:
df.groupby('Element').sum()

Area Code  Item Code  Element Code  \
Element                                                                      
Domestic supply quantity                   708993   14197445      28068795   
Export Quantity                            599910   11840553      26026133   
Fat supply quantity (g/capita/day)         675050   13535000       3435732   
Feed                                       176272    3538507       7282199   
Food                                       663295   13285035      25406622   
Food supply (kcal/capita/day)              674057   13511060       3329296   
Food supply quantity (kg/capita/yr)        658446   13185401       3163725   
Import Quantity                            688174   13795966      28834929   
Losses                                     274353    5424803      10292107   
Other uses (non-food)                      235554    4729749       8926728   
Processing                                 271940    5350416      10313310   
Production                                 526751   10450053      21388191   
Protein supply quantity (g/capita/day)     675050   13535000       3385502   
Residuals                                  623271   12421089      24066350   
Seed                                       103537    2035933       4211574   
Stock Variation                            571566   11329527      21464704   
Total Population - Both sexes                6020     112545         22995   
Tourist consumption                         50308    1486528       2869905   

                                             Y2014       Y2015       Y2016  \
Element                                                                      
Domestic supply quantity                1996716.35  2021493.55  2044842.70   
Export Quantity                          150020.64   157614.47   151920.46   
Fat supply quantity (g/capita/day)        10225.56    10235.74    10102.77   
Feed                                     216927.89   225050.22   228958.65   
Food                                    1212332.49  1232361.10  1247022.17   
Food supply (kcal/capita/day)            454257.00   453383.00   451810.00   
Food supply quantity (kg/capita/yr)       49650.63    49345.13    48985.28   
Import Quantity                          274144.48   267018.46   286582.78   
Losses                                   153223.00   155439.00   157787.00   
Other uses (non-food)                     78718.13    66254.41    69563.68   
Processing                               282923.00   287929.00   280631.00   
Production                              1931287.75  1947019.39  1943537.15   
Protein supply quantity (g/capita/day)    11836.46    11833.95    11779.69   
Residuals                                 30149.00    30045.00    37224.00   
Seed                                      21922.92    23976.82    23389.20   
Stock Variation                           58749.83    34910.99    33140.12   
Total Population - Both sexes           1031585.00  1058081.00  1085107.00   
Tourist consumption                         416.00      349.00       89.00   

                                             Y2017       Y2018  
Element                                                         
Domestic supply quantity                2088198.10  2161192.10  
Export Quantity                          182338.80   181594.80  
Fat supply quantity (g/capita/day)        10253.84    10258.69  
Feed                                     223705.68   233489.68  
Food                                    1258888.28  1303841.28  
Food supply (kcal/capita/day)            454681.00   455261.00  
Food supply quantity (kg/capita/yr)       48690.04    49056.85  
Import Quantity                          294559.09   287997.09  
Losses                                   160614.00   163902.00  
Other uses (non-food)                     91645.97    91300.97  
Processing                               292836.00   308429.00  
Production                              2030056.89  2075072.89  
Protein 

### Result:
    The total number of Production in 2014 is 1931287.75

## Which of these elements had the highest sum in 2018?

In [55]:
#select columns 'Y2018' and 'Element'

df_y2018_element = df[['Y2018','Element']]


In [63]:
df_y2018_element.groupby('Element').sum().sort_values(by='Y2018',ascending=False)

Y2018
Element                                           
Domestic supply quantity                2161192.10
Production                              2075072.89
Food                                    1303841.28
Total Population - Both sexes           1140605.00
Food supply (kcal/capita/day)            455261.00
Processing                               308429.00
Import Quantity                          287997.09
Feed                                     233489.68
Export Quantity                          181594.80
Losses                                   163902.00
Other uses (non-food)                     91300.97
Food supply quantity (kg/capita/yr)       49056.85
Residuals                                 34864.00
Seed                                      25263.14
Stock Variation                           20577.91
Protein supply quantity (g/capita/day)    11833.56
Fat supply quantity (g/capita/day)        10258.69
Tourist consumption                          90.00

### Result:
    Domestic supply quantity (2161192.10) is the "Element" that had the highest sum 2018.

## Which of these elements had the 3rd lowest sum in 2018?

In [64]:
df_y2018_element.groupby('Element').sum().sort_values(by='Y2018',ascending=True)

Y2018
Element                                           
Tourist consumption                          90.00
Fat supply quantity (g/capita/day)        10258.69
Protein supply quantity (g/capita/day)    11833.56
Stock Variation                           20577.91
Seed                                      25263.14
Residuals                                 34864.00
Food supply quantity (kg/capita/yr)       49056.85
Other uses (non-food)                     91300.97
Losses                                   163902.00
Export Quantity                          181594.80
Feed                                     233489.68
Import Quantity                          287997.09
Processing                               308429.00
Food supply (kcal/capita/day)            455261.00
Total Population - Both sexes           1140605.00
Food                                    1303841.28
Production                              2075072.89
Domestic supply quantity                2161192.10

### Result:
    Protein supply quantity (g/capita/day) (11833.56) is the "Element" that had the 3rd lowest sum in 2018

## What is the total Import Quantity in Algeria in 2018?

In [98]:
dfx = df[['Element', 'Area', 'Y2018']]

In [87]:
dfx.head()

Element     Area     Y2018
0           Total Population - Both sexes  Algeria  42228.00
1                Domestic supply quantity  Algeria      0.00
2           Food supply (kcal/capita/day)  Algeria   3322.00
3  Protein supply quantity (g/capita/day)  Algeria     91.83
4      Fat supply quantity (g/capita/day)  Algeria     77.28

In [106]:
dfx1 = dfx.groupby(['Element','Area']).sum()

In [110]:
dfx1

Y2018
Element                                Area                                  
Domestic supply quantity               Algeria                       90023.42
                                       Angola                        59590.22
                                       Benin                         31447.90
                                       Botswana                       3045.85
                                       Burkina Faso                  25705.16
                                       Cabo Verde                      731.00
                                       Cameroon                      64009.74
                                       Central African Republic       5567.06
                                       Chad                          15489.73
                                       Comoros                          19.95
                                       Congo                          8524.55
                                       C�te d'Ivoire                 55557.64
                                       Djibouti                       2760.96
                                       Egypt                        248477.72
                                       Eswatini                      13158.81
                                       Ethiopia                     107850.72
                                       Ethiopia PDR                      0.00
                                       Gabon                          4608.75
                                       Gambia                         2221.89
                                       Ghana                        100558.45
                                       Guinea                        25292.85
                                       Guinea-Bissau                  2049.58
                                       Kenya                         67202.58
                                       Lesotho                        2062.73
                                       Liberia                        4998.99
                                       Madagascar                    32596.87
                                       Malawi                        50937.37
                                       Mali                          40795.02
                                       Mauritania                    22053.08
                                       Mauritius                      8290.36
                                       Morocco                       86458.07
                                       Mozambique                    42581.75
                                       Namibia                        4019.66
                                       Niger                         31258.84
                                       Nigeria                      428323.32
                                       Rwanda                        17426.52
                                       Sao Tome and Principe           349.73
                                       Senegal                       19374.64
                                       Seychelles                       10.68
                                       Sierra Leone                  12753.68
                                       South Africa                 126493.86
                                       Sudan                         65451.77
                                       Sudan (former)                    0.00
                                       Togo                          10000.22
                                       Tunisia                       31347.90
                                       Uganda                        54617.10
                                       United Republic of Tanzania   92239.88
                                       Zambia                        24994.80
                                       Zimbabwe                      17860.73
Export Quantity                        Algeria                         811.54
                

### Result:
    The total Import Quantity in Algeria in 2018 is 36238.29

## What is the total number of unique countries in the dataset?

In [168]:
df['Area'].unique()

array(['Algeria', 'Angola', 'Benin', 'Botswana', 'Burkina Faso',
       'Cabo Verde', 'Cameroon', 'Central African Republic', 'Chad',
       'Comoros', 'Congo', "C�te d'Ivoire", 'Djibouti', 'Egypt',
       'Eswatini', 'Ethiopia', 'Ethiopia PDR', 'Gabon', 'Gambia', 'Ghana',
       'Guinea', 'Guinea-Bissau', 'Kenya', 'Lesotho', 'Liberia',
       'Madagascar', 'Malawi', 'Mali', 'Mauritania', 'Mauritius',
       'Morocco', 'Mozambique', 'Namibia', 'Niger', 'Nigeria', 'Rwanda',
       'Sao Tome and Principe', 'Senegal', 'Seychelles', 'Sierra Leone',
       'South Africa', 'Sudan', 'Sudan (former)', 'Togo', 'Tunisia',
       'Uganda', 'United Republic of Tanzania', 'Zambia', 'Zimbabwe'],
      dtype=object)

In [164]:
df['Area'].nunique()

49

### Result:
    The total number of unique countries in the dataset is 49